specs/progress.md

```diff
@@ -0,0 +1,142 @@
+# fast-progress
+
+## Overview
+
+*Progress* is used to display the length of a process or to represent an unspecified wait time. *Progress* can either be circular or linear. *Progress* can either be in an **indeterminate** or **determinate** state. This is determined by whether `value` attribute is passed to the component. 
+
+### Use Cases
+
+The *progress* component should be used for an element that displays the progress status for a task that takes a long time or consists of several steps. Typical use cases include loading elements or status of a download. 
+  
+### Features
+
+A progress should allow the following properties:
+- `value`, the value of the progress, if not passed the progress will be put into its "indeterminate" state.
+- `min-value`, the minimum value.
+- `max-value`, the maximum value.
+- `circular`, whether the progress renders circular or not.
+- `paused`, whether the progress is paused or not.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/progress)
+- [Material UI](https://material-ui.com/components/progress/)
+- [Lightning Design / Linear](https://www.lightningdesignsystem.com/components/progress-bar/)
+- [Lightning Design / Circular](https://www.lightningdesignsystem.com/components/progress-ring/)
+- [Ant Design](https://ant.design/components/progress/)
+- [Atlassian / Linear](https://atlaskit.atlassian.com/packages/server/progress-bar)
+- [Atlassian / Circular](https://atlaskit.atlassian.com/packages/core/spinner)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/progress-controls)
+
+---
+
+### API
+
+*Component name:*
+- `fast-progress`
+
+*Attributes:*
+- `value`: number
+- `min-value`: number
+- `max-value`: number
+- `circular`: boolean
+- `paused`: boolean
+
+### Anatomy and Appearance
+
+*Parts:*
+- determinate
+- indeterminate
+- circular-container
+- circular-background
+- circular-determinate
+- circular-indeterminate
+- linear-determinate
+- linear-indeterminate-indicator-1
+- linear-indeterminate-indicator-2
+
+*Circular Determinate Template:*
+```
+<div
+    role="progressbar"
```

Comment:
I vaguely remember there was a reason, but why not use the progress element here instead?

Reply:
In that case, can you investigate and add the reasoning into the spec?

Reply:
I believe that is why also, but it's been a while.